src/pip/_internal/index/package_finder.py

```diff
@@ -837,17 +1024,38 @@ def make_candidate_evaluator(
         # type: (...) -> CandidateEvaluator
         """Create a CandidateEvaluator object to use.
         """
-        candidate_prefs = self._candidate_prefs
-        return CandidateEvaluator.create(
-            project_name=project_name,
-            target_python=self._target_python,
-            prefer_binary=candidate_prefs.prefer_binary,
-            allow_all_prereleases=candidate_prefs.allow_all_prereleases,
-            specifier=specifier,
-            hashes=hashes,
+        package_finder_tuple = self.get_state_as_tuple()
+        return PackageFinder.make_candidate_evaluator_static(
+            package_finder_tuple,
+            project_name,
+            specifier,
+            hashes
         )
 
+    @staticmethod
     @functools.lru_cache(maxsize=None)
+    def find_best_candidate_static(
```

Comment:
It’d be clearer to name this a private function (adding a leading `_`). There are other functions that can be similarly changed to signify they are only internal and not called elsewhere (which improves maintainability).